cloudplatform/connectivity-apache-httpclient4/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/HttpClientWrapper.java

```diff
@@ -76,6 +77,18 @@ public String toString()
     HttpClientWrapper( final CloseableHttpClient httpClient, final HttpDestinationProperties destination )
     {
         this.httpClient = httpClient;
+
+        // check destination is configured correctly
+        // Any change here must be copied to DefaultHttpDestination.Builder.build()
+        if( destination.get(DestinationProperty.PROXY_TYPE).contains(ProxyType.ON_PREMISE)
+            && destination.get(DestinationProperty.PROXY_HOST).isEmpty() ) {
+
+            // throw the original error
+            new DefaultHttpDestinationBuilderProxyHandler().handle(DefaultHttpDestination.fromDestination(destination));
+            // just in case the service binding has changed and the exception does not occur anymore
+            throw new DestinationAccessException(
+                "Unable to resolve connectivity service binding. Please check the logs.");
+        }
```

Comment:
Let's stick what we've aligned on:

```suggestion

        if( destination.getProxyType().contains(ProxyType.ON_PREMISE)
            && destination.getProxyConfiguration().isEmpty() ) {
            throw new DestinationAccessException(
                "Unable to create an HttpClient from the provided destination. "
                    + "The destination is supposed to target an on-premise system but lacks the correct proxy configuration. "
                    + "Please check the application logs for further details.");
        }
```

---

Also: Please apply the same change to the `HttpClient5Wrapper`.